lib/pushy_client/protocol_handler.rb

```diff
@@ -413,10 +406,15 @@ def check_server_address(name, server_address)
     def validate_params(params = {})
       stdout_bytes = params[:stdout].to_s.bytesize
       stderr_bytes = params[:stderr].to_s.bytesize
-
-      if (stdout_bytes + stderr_bytes) > MAX_BODY_SIZE
+      if (stdout_bytes + stderr_bytes) > client.max_body_size.to_i
+        Chef::Log.warn("****************************************************")
         Chef::Log.warn("Command output too long. Will not be sent to server.")
+        Chef::Log.warn("****************************************************")
         params.delete_if { |k| [:stdout, :stderr].include? k }
+      elsif client.max_body_size.to_i > @max_body_size
+        Chef::Log.warn("****************************************************")
+        Chef::Log.warn("Client MAX_BODY_SIZE is larger than server.")
```

Comment:
I think we should also add the server current value and client current value to the warning message when the client value is larger than the server. IE: "Client MAX_BODY_SIZE of XXX is larger than the server MAX_BODY_SIZE of XXX" or something to that effect. However, if you think this is just going to be more confusing though we can leave it as is.